ksql-metastore/src/main/java/io/confluent/ksql/metastore/StructuredDataSource.java

```diff
@@ -76,6 +76,10 @@ public KsqlTopicSerDe getKsqlTopicSerde() {
     return ksqlTopic.getKsqlTopicSerDe();
   }
 
+  public boolean isAvroSerialized() {
+    return getKsqlTopicSerde().getSerDe() == DataSource.DataSourceSerDe.AVRO;
```

Comment:
I think that particular example is fine since `QueryEngine#getResultDatasource` is only ever called in a sandbox context that doesn't actually execute commands like DropSource or TerminateCluster for which we're interested in checking the serde format of the data source. Nonetheless, you bring up a great point and an extra null check never hurt anyone, so I've added it. Thanks.